Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -155,16 +155,76 @@ final class TestDiscoveryCommand: CustomLLBuildCommand {
 
         stream <<< "import XCTest" <<< "\n\n"
 
+        stream <<< "@available(*, deprecated, message: \"Not actually deprecated. Marked as deprecated to allow inclusion of deprecated tests (which test deprecated functionality) without warnings\")" <<< "\n"
+        stream <<< "public func __allModuleTests() -> [XCTestCaseEntry] {" <<< "\n"
```

Comment:
not sure about the name. iiuc we are creating a new module now with all the discovered XCTests and this function exposes them?

Reply:
>iiuc we are creating a new module now with all the discovered XCTests and this function exposes them?

Exactly, yeah. Currently, SwiftPM synthesizes code (elsewhere) which finds all the tests in the **module** of each test target in the package, and for each one, it defines a function returning that test module's array of tests. The API signature of those functions is:
```swift
func __\(moduleName)__allTests() -> [XCTestCaseEntry]
```
And those, in turn, return an array of their module's tests, which consist of the `__allTests_\(className)` property for each class:
```swift
    stream <<< indent(8) <<< "testCase(\(className).__allTests__\(className)),\n"
```
In both of those examples, I noticed the common usage of double-underscore prefixes, which I think were chosen to prevent the possibility of name collisions with real code in the test target module. I think this was particularly important for the `__allTests__\(className)` property because it's in an `extension` on the actual test class. It may not technically be necessary for us to use underscore prefixes on the global `func`'s we define elsewhere, though.

So I'm open to alternatives on this `func __allModuleTests()` name, but that's some context on why I picked that — basically aiming for consistency with the existing code.

Reply:
I see. would `__allDiscoveredTests ` be clearer? what is the

Reply:
The new module name containing the discovered tests is `<PackageName>DiscoveredTests`.
The manifest target still has its prior name, `<PackageName>Tests`.

I'd be fine with `func __allDiscoveredTests()`, yeah!

Reply:
😎